service/ec2/customizations.go

```diff
@@ -4,19 +4,41 @@ import (
 	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
-	request "github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/internal/awsutil"
 )
 
+const (
+	// customRetryerMaxNumRetries sets max number of retries
+	customRetryerMaxNumRetries = 3
+
+	// customRetryerMinRetryDelay sets min retry delay
+	customRetryerMinRetryDelay = 1 * time.Second
+
+	// customRetryerMaxRetryDelay sets max retry delay
+	customRetryerMaxRetryDelay = 8 * time.Second
+)
+
+// setCustomRetryer overides the default Retryer values
+func setCustomRetryer(d *aws.DefaultRetryer) {
+	d.NumMaxRetries = customRetryerMaxNumRetries
+	d.MinRetryDelay = customRetryerMinRetryDelay
+	d.MinThrottleDelay = customRetryerMinRetryDelay
+	d.MaxRetryDelay = customRetryerMaxRetryDelay
+	d.MaxThrottleDelay = customRetryerMaxRetryDelay
+}
+
 func init() {
-	initRequest = func(c *Client, r *request.Request) {
+	initRequest = func(c *Client, r *aws.Request) {
 		if r.Operation.Name == opCopySnapshot { // fill the PresignedURL parameter
 			r.Handlers.Build.PushFront(fillPresignedURL)
 		}
+		if c.Config.Retryer == nil && (r.Operation.Name == opModifyNetworkInterfaceAttribute || r.Operation.Name == opAssignPrivateIpAddresses) {
+			r.Retryer = aws.Retryer(aws.NewDefaultRetryer(setCustomRetryer))
```

Comment:
Shouldn't need to case the returned value of `aws.NewDefaultRetryer` to `aws.Retryer`